clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -517,8 +520,9 @@ PointerVariableConstraint::PointerVariableConstraint(
   // Get a string representing the type without pointer and array indirection.
   BaseType = extractBaseType(D, TSInfo, QT, Ty, C);
 
-  IsVoidPtr = isTypeHasVoid(QT);
-  bool IsWild = !getIsGeneric() && (isVarArgType(BaseType) || IsVoidPtr);
+  IsVoidPtr = isPtrOrArrayType(QT) && isTypeHasVoid(QT);
```

Comment:
Why change `isVoidPtr` to specifically checked `isPtrOrArrayType(QT)` when it wasn't being checked before? Is `QT` the entire type in question, or just the base type of what we already understand to be a pointer type?

Reply:
`QT` is the base type extracted above. This was needed because a return type of `void` (no ptr) was marked as potentially a generic, and this fixed it. I didn't look beyond that. It may be a bug I'm looking for, and I'll have to fix the return type bug in another way.

Reply:
My description above was incorrect, `QT` is the original type passed to the constructor. `QTy` (unused here) is the extracted type that should now be the base. So, `IsVoidPtr` was previously `true` for `void`(no ptr) types, and I assume this was in error.